### PR TITLE
fix jax jit with broadcasted probs

### PR DESCRIPTION
**Pull request by albi3ro**
Fixes Issue #4738 

The `ProbabilityMP.shape` did not take into account probabilities that broadcasted across all available wires.

[sc-48626]

### CODE REVIEW

File: doc/releases/changelog-0.33.0.md [hunk 1]
```diff
@@ -537,6 +537,8 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Jax jit now works when a probability measurement is broadcasted onto all wires.
```
Comment (albi3ro) [inline doc/releases/changelog-0.33.0.md:540]:
```suggestion
* Jax jit now works when a probability measurement is broadcasted onto all wires.
  [(#4742)](https://github.com/PennyLaneAI/pennylane/pull/4742)
```

**Review by mudit2812 — COMMENTED** [review]
This problem exists for `sample().shape` and `state().shape` as well. I think it's worthwhile updating them in this PR as well.

**Comment by albi3ro** [comment]
> This problem exists for `sample().shape` and `state().shape` as well. I think it's worthwhile updating them in this PR as well.

jax-jit does not work with `sample` or `state` anyway, but it looks like they don't have the problem:
```
>>> qml.state().shape(qml.device('default.qubit', wires=(0,1)), qml.measurements.Shots(None))
(4,)
>>> qml.sample().shape(qml.device('default.qubit', wires=(0,1)), qml.measurements.Shots(5))
(5, 2)
```

**Comment by mudit2812** [comment]
> jax-jit does not work with sample or state anyway, but it looks like they don't have the problem:

Sorry, I meant for cases when devices without wires are used.

**Comment by albi3ro** [comment]
> > jax-jit does not work with sample or state anyway, but it looks like they don't have the problem:
> 
> Sorry, I meant for cases when devices without wires are used.

For devices without wires we are going to need to make a breaking change to the call signature of `MeasurementProcess.shape`. We will need to change from `(device, shots)` to `(wires, shots)` as the wires may be coming from the tape instead of from the device.

**Review by mudit2812 — APPROVED** [review]
Thanks!

**Comment by timmysilv** [comment]
@albi3ro we might not need a breaking change. I used to just replace `device` with some object that also had wires defined (at the time, it was `self`) but we can do anything. I added it early in the device port, but I removed it from qscript.py [here](https://github.com/PennyLaneAI/pennylane/commit/036102de9a23a6d3ea68d5709b7606b925d650a2#diff-fe39c5ae8018118cba5abadd13d54040cee7f520767ebf3459dec817ec8a2a91)

**Pull request merged**